tests/testthat/test-3agglomerate.R

```diff
@@ -153,13 +153,53 @@ test_that("agglomerate", {
     expect_true( !any( duplicated(rownames(uniq)) ) )
     expect_true( any( duplicated(rownames(not_uniq)) ) )
     
+    ## START test agglomerateByModule ##
+    
+    # Generate random modules
+    N_module <- 30L
+    modules <- sample(
+        c(TRUE, FALSE),
+        size = nrow(tse) * N_module,
+        prob = c(0.2, 0.8),
+        replace = TRUE
+    )
+    # Add modules to rowData
+    modules <- modules |>
+        matrix(nrow = nrow(tse))
+    # Add row and column names
+    rownames(modules) <- rownames(tse)
+    colnames(modules) <- paste0("module_", seq_len(ncol(modules)))
+    # Store modules in metadata slot
+    metadata(tse)$modules <- modules
+    # Add pseudocount assay
```

Comment:
Would be great if we had a demo data set that is already in correct format so that it would be possible to have a minimal example that only focuses on the feature being tested?

Reply:
Yes but it might be easier after we've developed some method to fetch modules from a database.

Reply:
We do have butyrate producing taxa for Metaphlan (3 or 4?) and for Greengenes2. We could use that as an example (this is just one module, though) if we have a gut microbiome example data set with one of these methods. But OK to wait then if not.

Reply:
I think we could add an example with a mia dataset and signatures from BugSigDB.